### PR TITLE
chore: [IOAPPCIT-24] Add a crypto key pair generation test at application startup.

### PR DESCRIPTION
**Pull request by shadowsheep1**
## Short description
This PR adds a crypto key pair generation test at application startup using [io-react-native-crypto](https://github.com/pagopa/io-react-native-crypto) library.

## List of changes proposed in this pull request
- ⚠️ .circleci/config.yml: bumps `node` version from `14.17.4` to `16.19.0`.
- ⚠️ .node-version: bumps `node` version from `14.17.4` to `16.19.0`.
- ⚠️ android/build.gradle: bumps `kotlin` version from `1.5.21` to `1.7.0`.
- yarn.lock: adds `io-react-natiive-crypto` library.
- ios/Podfile.lock: adds `io-react-natiive-crypto` library pod.
- package.json: adds `io-react-natiive-crypto` library.
- ts/sagas/startup.ts: adds the key pair generation.
- ts/sagas/startup/generateCryptoKeyPair.ts: this is the new key generation saga.
- ts/store/reducers/__mock__/backendStatus.ts: update the mock with the new definitions.
- ts/store/reducers/backendStatus.ts: add the selector for the new feature flag to enable/disable the key generation.
- ts/utils/crypto.ts: utility functions for the `io-react-natiive-crypto` library.

## How to test
Run the application on both Android and iOS physical devices against the [io-dev-api-server](https://github.com/pagopa/io-dev-api-server). Try running the application also on simulators: it's not guaranteed that the key generation will be successful, but also a crash-free unsuccessful event is good. 
You must set to `true` this [flag](https://github.com/pagopa/io-dev-api-server/blob/c9485701686abc0f47c4272f05d6fd5246a6cd7e/src/payloads/backend.ts#L78) on [io-dev-api-server](https://github.com/pagopa/io-dev-api-server) to enable the key generation.
Try with different scenarios:
- Try with the flag above in a disabled state with a first onboarding user. No key generation should happen.
- Then enable the flag and restart the application. You should see a key generation attempt.
- Try with the flag above in an enabled state with an already logged-in user: you should see a key generation attempt.
- More up to you ;)

### CODE REVIEW

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (hevelius) [inline ts/sagas/startup/generateCryptoKeyPair.ts:10]: Why did you use this value? *-temp-*

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (hevelius) [inline ts/sagas/startup/generateCryptoKeyPair.ts:16]: Maybe a more explicit name could be used. eg. `isKeyExist` or `isKeyEmpty`

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (hevelius) [inline ts/sagas/startup/generateCryptoKeyPair.ts:24]: What happen here with deleteKey?

**Review by hevelius — COMMENTED** [review]
Is there a local FF for lollipop or only remote?

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (shadowsheep1) [reply ts/sagas/startup/generateCryptoKeyPair.ts:10]: Because we create the key to test the functionality and then we remove it. In a near future refactoring we will also give the key a more valuable name.

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (shadowsheep1) [reply ts/sagas/startup/generateCryptoKeyPair.ts:16]: Do you mean on line # 16 or the first two lines or all the lines? Usually if you use `is` you don't use `exists`.  Those are convention mutually exclusive. But I agree with you that we could improve the names of this flags. 

What do you think of the below proposal?

```ts
const keyAlreadyExistsOnKeystore = yield* call(checkPublicKeyExists, KEY_NAME);
const keyHasBeenAlreadyGenerated = yield* call(isKeyAlreadyGenerated, KEY_NAME);
const shouldWeGenerateKey = !(keyHasBeenAlreadyGenerated || keyAlreadyExistsOnKeystore);
```

File: ts/sagas/startup/generateCryptoKeyPair.ts
Comment (shadowsheep1) [reply ts/sagas/startup/generateCryptoKeyPair.ts:24]: The key is removed from the keystore after it's been generated successfully. This is only a test. We leave nothing on the device after this test.

**Comment by shadowsheep1** [comment]
> Is there a local FF for lollipop or only remote?

@hevelius only a remote one.

File: ts/store/reducers/backendStatus.ts
Comment (hevelius) [inline ts/store/reducers/backendStatus.ts:225]: It's not an error but just curiosity why not use `O.gerOrElseW` instead `O.toUndefined` removing nullish op

**Review by hevelius — APPROVED** [review]
LGTM

File: ts/store/reducers/backendStatus.ts
Comment (shadowsheep1) [reply ts/store/reducers/backendStatus.ts:225]: Good point! Just because I copied from an existing one tbh :)
I would say that here should be enough `O.getOrElse` because the type returned is the same:

```ts
export const isLollipopEnabledSelector = createSelector(
  backendStatusSelector,
  (backendStatus): boolean =>
    pipe(
      backendStatus,
      O.map(bs => bs.config.lollipop.enabled),
      O.getOrElse(() => false)
    )
);
```

**Comment by shadowsheep1** [comment]
@CaraDune can we go ahead with this PR, having approved this #4296? Waiting for your approval ;) 💪

**Comment by CaraDune** [comment]
@shadowsheep1 go ahead, please and thank you!

**Review by hevelius — APPROVED** [review]

File: package.json
Comment (CrisTofani) [inline package.json:136]: Since we released the package on NPM can we point the official release? 😄

**Review by CrisTofani — CHANGES_REQUESTED** [review]

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**
